### ISSUE TITLE
use an external library for all multivar. algorithms in normal.cpp

### ISSUE THREAD

**Issue by rwst** [pynac/pynac#2]
```
Burcin Erocal created an issue 2011-05-09

The GCD implementation for multivariate polynomials provided in Sage, through Singular,
is much more capable than the one from GiNaC. We should change ginac/normal.cpp
to call a Python function for this purpose and handle the rest in Sage.
```
Transferred from http://hg.pynac.org/pynac/issue/1/use-sage-for-gcd

And this is from William or Burcin, https://wiki.sagemath.org/symbolics/pynac_todo
```
Maybe change SAGE's Ginac to make a call to a cython gcd function, then use
singular, since singular's gcd over QQ is much better than ginac's, I think,
and ginac *only* does GCD over QQ.  Actually, just make everything in normal.cpp
be implemented via Singular, probably...
```

**Comment by rwst** [comment]
Actually from all 500 files in `src/sage/symbolic`, `src/sage/calculus`, `src/doc` only 38 doctests in `src/sage/symbolic/expression.pyx` use the GiNaC `gcd`.

**Comment by rwst** [comment]
(Mail with Bernard Parisse from 2016-May-10)
```
> Bonjour Bernard,
>
> I'm the current maintainer of Sage's symbolic library Pynac.
> Would you recommend using the Giac library for those
> multivar. polynomial manipulations that typically come up in symbolics,
> like gcd? Is it faster for such manipulations than SIngular?
>
> Regards,
Hi,

Yes, I believe that giac multivariate polynomial code
(multiplication/gcd/factorization) is faster than singular. In fact, I
did not check myself with singular, I just looked at
https://wiki.sagemath.org/MultivariateGCDBenchmarks
and compared to my own (now old) benchmarks
http://www-fourier.ujf-grenoble.fr/~parisse/giac/benchmarks/benchmarks.html
and concluded that the real competition was not with singular but more
with Roman Pearce code in maple or Allan Steel in magma (this might be
wrong if singular has done recent progress in this area, but I didn't
find time to check myself).
```

**Comment by rwst** [comment]
Resolved in 27c6a541e2e035b3445d39776da0a1ae35610d8f
